source/BrightcoveOS .NET-MAPI-Wrapper/Api/BrightcoveApi.video.write.cs

```diff
@@ -430,18 +430,86 @@ private BrightcoveImage DoAddImage(BrightcoveImage image, FileUploadInfo fileUpl
 			return RunFilePost<BrightcoveResultContainer<BrightcoveImage>>(parms, fileUploadInfo).Result;
 		}
 
-		#endregion
-
-		#region AddLogoOverlay
-
-		/// <summary>
-		/// Adds a logo overlay image to a video.
-		/// </summary>
-		/// <param name="logoOverlay">The metadata for the logo overlay you'd like to create (or update).</param>
-		/// <param name="fileUploadInfo">Information for the file to be uploaded.</param>
-		/// <param name="videoId">The ID of the video you want to assign a logo overlay to.</param>
-		/// <returns>The newly created or updated BrightcoveLogoOverlay</returns>
-		public BrightcoveLogoOverlay AddLogoOverlay(BrightcoveLogoOverlay logoOverlay, FileUploadInfo fileUploadInfo, long videoId)
+        #endregion
+
+        #region AddCaptioning
+        public BrightcoveCaptioning AddCaptioning(string captionFileUrl, long videoId, string filename)
+        {
+            return DoAddCaptioningWithExternalUrl(captionFileUrl, videoId, null, filename);
+        }
+
+        public BrightcoveCaptioning AddCaptioning(string captionFileUrl, string videoReferenceId, string filename)
+        {
+            return DoAddCaptioningWithExternalUrl(captionFileUrl, -1, videoReferenceId, filename);

```

Comment:
@kish9nine told me we will only be using videoId, but I added them both to comply with existing conventions